app/api/download/[file_name]/route.ts

```diff
@@ -0,0 +1,40 @@
+import path from "path";
+import fs from "fs";
+
+export const GET = async (
+  _req: Request,
+  { params: { file_name: fileName } }: { params: { file_name: string } }
+) => {
+  if (!fileName) {
+    return new Response("File not found", { status: 404 });
+  }
+  const decodedFileName = path.basename(decodeURIComponent(fileName)).trim();
+  // Ensure that the decodedFileName does not contain any directory traversal characters
+  if (decodedFileName.includes("..")) {
+    return new Response("Invalid file name", { status: 400 });
+  }
+  const filePath = path.join(process.cwd(), `./files/${decodedFileName}`);
+
+  try {
+    await fs.promises.access(filePath, fs.constants.R_OK);
+  } catch (err: any) {
+    if (err.code === "ENOENT") {
+      return new Response("File not found", { status: 404 });
+    } else {
+      console.error("Error accessing file:", err.message);
+      // Handle other possible errors (e.g., permission issues)
+      return new Response("Error accessing file", { status: 500 });
+    }
```

Comment:
**Suggestion:** Use 'fs.promises.stat' instead of 'fs.promises.access' to check if the file exists. 'fs.promises.stat' provides more information about the file, which could be useful in future enhancements.
```suggestion
  try {
    await fs.promises.stat(filePath);
  } catch (err: any) {
    if (err.code === "ENOENT") {
      return new Response("File not found", { status: 404 });
    } else {
      console.error("Error accessing file:", err.message);
      // Handle other possible errors (e.g., permission issues)
      return new Response("Error accessing file", { status: 500 });
    }
  }
```